### PR TITLE
PHP 8.4: CSV related changes and deprecations

### PR DESCRIPTION
**Pull request by Girgias**
I *might* have pushed it a bit with the XIncludes and XPath queries.

### CODE REVIEW

File: reference/filesystem/functions/fgetcsv.xml [hunk 1]
```diff
@@ -96,6 +97,14 @@
         even meant to escape itself.
        </simpara>
       </note>
+      <warning>
+       <simpara>
+        As of PHP 8.4.0, depending on the default value of
+        <parameter>escape</parameter> is deprecated.
+        It needs to be provided explicitly either positionally or by the use
+        of <link linkend="functions.named-arguments">Named Arguments</link>.
```
Comment (cmb69) [inline reference/filesystem/functions/fgetcsv.xml:105]: Not sure if we should use title case here.

**Review by cmb69 — COMMENTED** [review]
> I _might_ have pushed it a bit with the XIncludes and XPath queries.

Yeah, that looks somewhat fragile (e.g. relying on the parameter numbers). Might be better to use some entities here.

File: reference/strings/functions/str-getcsv.xml [hunk 2]
```diff
@@ -104,6 +75,19 @@
...
+      <xi:include xpointer="xmlns(db=http://docbook.org/ns/docbook) xpointer(id('function.fgetcsv')/db:refsect1[@role='changelog']//db:row[db:entry[text()='8.4.0']]/.)">
+       <xi:fallback/>
+      </xi:include>
+      <row>
+       <entry>8.4.0</entry>
+       <entry>
+        Now throws a <exceptionname>ValueError</exceptionname> if
+        <parameter>separator</parameter>, <parameter>enclosure</parameter>,
+        or <parameter>escape</parameter> is invalid.
+        This mimics the behaviour of <function>fgetcsv</function> and
```
Comment (cmb69) [inline reference/strings/functions/str-getcsv.xml:87]:
```suggestion
        This mimics the behavior of <function>fgetcsv</function> and
```

**Review by cmb69 — APPROVED** [review]
I'm not a fan of xincludes, but this looks quite reasonable. Minor nit below.

**Comment by Girgias** [comment]
I prefer XIncludes to Entities as `language-snippets` is a pain as a translator IMHO.

**Pull request merged**